src/pages/StudentCourses.js

```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+import { CourseCards, CourseCardsWithCert } from "../components/Card";
+import { useCourseData, useAttemptData } from "../components/FetchCourses";
+import { useContext } from "react";
+import { AuthContext } from "./AuthProvider";
+
+const AttemptedCourses = ({ userEmail }) => {
+  const { currentUser } = useContext(AuthContext);
+  const { attemptedCourseMap, isLoading } = useAttemptData({ userEmail });
+  return isLoading ? (
+    <span className="loading loading-dots loading-lg place-self-center sm:col-start-4 sm:col-span-1"></span>
+  ) : (
+    <>
+      <h3 className="sm:col-span-7">Completed</h3>
+      <CourseCardsWithCert
+        courseMap={attemptedCourseMap}
+        displayName={currentUser.displayName}
+      />
+    </>
+  );
```

Comment:
nice! I would break it up into 2 return statements!

```suggestion
if (isLoading) return <span className="loading loading-dots loading-lg place-self-center sm:col-start-4 sm:col-span-1"></span>;
 
 return (
    <>
      <h3 className="sm:col-span-7">Completed</h3>
      <CourseCardsWithCert
        courseMap={attemptedCourseMap}
        displayName={currentUser.displayName}
      />
    </>
  );
```